docs/quick-start.md

```diff
@@ -0,0 +1,136 @@
+# DAG Factory Quick Start Guide
+
+**DAG Factory** is a Python library [Apache Airflow®](https://airflow.apache.org) that simplifies DAG creation using declarative YAML configuration files.
+
+# Prerequisites
+
+The minimum requirements for **dag-factory** are:
+
+- Python 3.8.0+
+- [Apache Airflow®](https://airflow.apache.org) 2.0+
+
+# Step 1: Create a Virtual Environment
+
+First, create and activate a virtual environment:
+
+```commandline
+python3 -m venv dagfactory_env
+source dagfactory_env/bin/activate
+```
+
+# Step 2: Install Apache Airflow
+
+Next, you'll need to install [Apache Airflow®](https://airflow.apache.org). Follow these steps:
+
+1. Create a directory for your project and navigate to it:
+
+    ```commandline
+    mkdir dag-factory-quick-start && cd dag-factory-quick-start
+    ```
+
+2. Set the `AIRFLOW_HOME` environment variable:
+
+    ```commandline
+    export AIRFLOW_HOME=$(pwd)
+    export AIRFLOW__CORE__LOAD_EXAMPLES=False
+    ```
+
+3. Install Apache Airflow:
+
+    ```commandline
+    pip install apache-airflow
+    ```
+
+# Step 3: Install DAG Factory
+
+Now, install the DAG Factory library in your virtual environment:
+
+```commandline
+pip install dag-factory
+```
+
+# Step 4: Set Up the DAGS Folder
+
+Create a dags folder inside the $AIRFLOW_HOME directory, which is where your DAGs will be stored:
+
+```commandline
+mkdir dags
+```
+
+# Step 5: Define a DAG in YAML
+
+**DAG Factory** uses YAML files to define DAG configurations. Create a file named `example_dag_factory.yml` in the `$AIRFLOW_HOME/dags` folder with the following content:
+
+```yaml
+default:
+  default_args:
+    catchup: false,
+    start_date: 2024-11-11
+
+example_dag:
+  default_args:
+    owner: "custom_owner"
+  description: "this is an example dag"
+  schedule_interval: "0 3 * * *"
+  render_template_as_native_obj: True
+  tasks:
+    task_1:
+      operator: airflow.operators.bash_operator.BashOperator
+      bash_command: "echo 1"
+    task_2:
+      operator: airflow.operators.bash_operator.BashOperator
+      bash_command: "echo 2"
+      dependencies: [task_1]
+    task_3:
+      operator: airflow.operators.bash_operator.BashOperator
+      bash_command: "echo 2"
+      dependencies: [task_1]
+```
```

Comment:
Let's use snippets as introduced in #328; you can see an example of how to use them in #327. It will be safer if we reference an example that is being run/tested within our CI.